crates/oxc_minifier/src/ast_passes/peephole_fold_constants.rs

```diff
@@ -157,8 +157,19 @@ impl<'a> PeepholeFoldConstants {
                 Some(ctx.ast.move_expression(&mut expr.argument))
             }
             // `+1` -> `1`
-            UnaryOperator::UnaryPlus if expr.argument.is_number() => {
-                Some(ctx.ast.move_expression(&mut expr.argument))
+            UnaryOperator::UnaryPlus => {
+                if expr.argument.is_number() {
```

Comment:
Try using match if there are many if elses, we may have more branches in the future.

match xxx {
  _ if expr.argument.is_number() => {}
}